### PR TITLE
sdjournal: add GetDataBytes() and GetDataValueBytes() to Journal

### PR DESCRIPTION
**Pull request by flyingmutant**
This allows to avoid extra `string` → `[]byte` conversion overhead for cases
when journal field value is e.g. parsed (which usually requires `[]byte`).

### CODE REVIEW

File: sdjournal/journal.go [hunk 1]
```diff
@@ -651,6 +652,43 @@ func (j *Journal) GetDataValue(field string) (string, error) {
 	return strings.SplitN(val, "=", 2)[1], nil
 }
 
+// GetDataBytes gets the data object associated with a specific field from the
+// current journal entry.
+func (j *Journal) GetDataBytes(field string) ([]byte, error) {
+	sd_journal_get_data, err := j.getFunction("sd_journal_get_data")
```
Comment (jonboulle) [inline sdjournal/journal.go]: I'm tempted to say this is an overoptimisation (although I do think that the interface should probably have been `[]byte` in the first place). In any case, if we do add this, the code is 99% identical between the functions - can you factor it out so they can share it?

Comment (flyingmutant) [inline sdjournal/journal.go]: Yeah, of course, will do. I was just not sure which way you prefer (with common code extracted or with more linear code).

**Comment by flyingmutant** [comment]
I've factored the common code to the `Journal.getData()` helper.

**Comment by jonboulle** [comment]
code seems okay. @lucab thoughts?

**Comment by lucab** [comment]
I think it is ok, just two additional comments for @flyingmutant:
 * can you please squash the two commits together?
 * I'm trying to revamp the testsuite (see #179). Can you please add a test which exercises the whole GetDataValueBytes/GetDataValue/GetDataBytes/GetData chain?

**Comment by flyingmutant** [comment]
I've squashed the commits. Once #179 is merged, I'll add the tests.

**Comment by lucab** [comment]
Great, thanks! You can already add the tests anyway; even if not run by travis, we use them for local sanity checks.

**Comment by lucab** [comment]
@flyingmutant any news on this?

**Comment by flyingmutant** [comment]
@lucab thanks for the ping — I'll try to finish the PR in the next couple of days.

**Comment by lucab** [comment]
@flyingmutant thanks, take you time as I didn't mean to push, just wanted to know if there was any progress (for release-scheduling purposes).

**Comment by flyingmutant** [comment]
@lucab I've added the tests — but, as you can see, they do not pass. It seems that current master branch (4fdcc47482107227bd1ed258e5b7bfbc82c9512a) does not pass the tests, too, with the same reason:

```
--- FAIL: TestJournalGetEntry (0.00s)
	journal_test.go:220: Error reading to journal: EOF
```

I hope that once master will pass the tests, this branch will, too — but right now I have no time to dig into the reason of master's failure, unfortunately.

**Comment by lucab** [comment]
That issue has been addressed by #181 (just merged). If you rebase on top of it and push again, travis should be happy now.

**Comment by flyingmutant** [comment]
@lucab thanks! Tests pass now.

**Comment by lucab** [comment]
@flyingmutant Nice! Then just squash all commits together and I'll have a final pass on this.

**Comment by flyingmutant** [comment]
@lucab done.

**Comment by lucab** [comment]
LGTM, thanks a lot for your contribution and your patience going back-and-forth :+1:

**Pull request merged**